File: python/samples/02-agents/providers/copilot_studio.py
# Copyright (c) Microsoft. All rights reserved.

import asyncio

from agent_framework.microsoft import CopilotStudioAgent

"""
Copilot Studio Provider

Demonstrates setting up CopilotStudioAgent and running a simple query.
Configuration is read from environment variables automatically.

Environment variables:
- COPILOTSTUDIOAGENT__ENVIRONMENTID: Environment ID
- COPILOTSTUDIOAGENT__SCHEMANAME: Agent schema name
- COPILOTSTUDIOAGENT__AGENTAPPID: Client ID for authentication
- COPILOTSTUDIOAGENT__TENANTID: Tenant ID for authentication

For more Copilot Studio examples:
- Explicit settings: getting_started/agents/copilotstudio/copilotstudio_with_explicit_settings.py
- Docs: https://learn.microsoft.com/agent-framework/providers/copilot-studio
"""


async def main() -> None:
    print("=== Copilot Studio Provider ===\n")

    # <create_agent>
    agent = CopilotStudioAgent()
    # </create_agent>

    # <run_query>
    query = "What is the capital of France?"
    print(f"User: {query}")
    result = await agent.run(query)
    print(f"Agent: {result}")
    # </run_query>


if __name__ == "__main__":
    asyncio.run(main())

File: dotnet/samples/02-agents/providers/CopilotStudio.cs
// Copyright (c) Microsoft. All rights reserved.

// Provider: Copilot Studio
// Create an agent backed by a Microsoft Copilot Studio agent.
// Note: Copilot Studio agents are accessed via the A2A protocol.
//
// For docs: https://learn.microsoft.com/agent-framework/agents/providers

using A2A;
using Microsoft.Agents.AI;

var a2aAgentHost = Environment.GetEnvironmentVariable("A2A_AGENT_HOST")
    ?? throw new InvalidOperationException("A2A_AGENT_HOST is not set.");

// <copilot_studio>
// Connect to a Copilot Studio agent via its A2A endpoint
A2ACardResolver agentCardResolver = new(new Uri(a2aAgentHost));
AIAgent agent = await agentCardResolver.GetAIAgentAsync();
// </copilot_studio>

Console.WriteLine(await agent.RunAsync("Tell me a joke about a pirate."));
